### ISSUE TITLE
can not cancel "switch network" => can not access account list any more

### ISSUE THREAD

**Issue by SvenMeyer** [MetaMask/metamask-extension#12052]
- went to [solarbeam.io](solarbeam.io)
- switch network pops up
- I decided that I do not want to switch network, click [cancel] , does not work
- go to another website / blankpage, can still not cancel "switch network"
- can not even list my accounts, as all I see is the "switch network" tab

![Metamask-switch-network](https://user-images.githubusercontent.com/25609/132645146-91b987f7-dbec-4286-9ccf-2dc50a0c7c52.png)

**Comment by danjm** [comment]
@SvenMeyer Can you provide some more details.

Which browser are you using?
Which version of MetaMask are you using?
What network were you on when you first went to solarbeam.io?

**Comment by imsys** [comment]
Seems similar to #11799

**Comment by imsys** [comment]
I just did a test on the same website you mentioned, and it worked fine, no problems. But I'm having the same problem in Firefox.

Chromium on Linux, Version 92.0.4515.107
MetaMask Version : 10.1.0

**Comment by imsys** [comment]
Upgraded to Chromium 93.0.4577.82, and still no problem there.

**Comment by daemianmack** [comment]
I've been suffering this problem in Firefox on OSX. It only happens in Firefox -- not Chrome or Brave, all of which I use frequently.

The problem is actually worse than the opening comment on this issue implies, because some networks (Fantom is the biggest offender) are aggressive about switching you back to their network _even when you're on a different tab_ trying to use some other network. In this case, you cannot interact with the app because a background tab is requesting you switch networks, and you cannot cancel the dialog. 

The only workaround is go find all the tabs on the requesting network, kill them, and reload the page you want to interact with. This can happen at any time, even in the middle of a series of transactions.

**Comment by SvenMeyer** [comment]
@imsys mhh, now that I am reading your comment, I would not rule out that I actually was using Firefox, not Chrome :-|

**Comment by imsys** [comment]
> @imsys mhh, now that I am reading your comment, I would not rule out that I actually was using Firefox, not Chrome :-|

@SvenMeyer so it's probably Firefox, I just compared the top-right bar, and your screenshot has the menu button of Firefox. In Chrome and Chromium it is a bit different.

**Comment by michaelday008** [comment]
I have this issue and it's a real pain.

Latest Metamask, latest firefox, Windows 10.

Here's how I reproduce it.

1. Open up some tab that uses a specific network like beefy finance and ensure you are on the correct network
2. Open up another tab that uses a different network like synapse protocol
3. Switch to the different network
4. Now suddenly beefy tries to switch you back to Binance
5. Get annoyed with beefy trying to switch and close the beefy tab
6. Now click the cancel button... Nothing happens
7. Click the switch network button, nothing happens
8. Cannot view account list because the switch network dialog removes all other options (no ... menu etc)

The only way I found out of this is to open up yet another tab, and go to some other site that uses a different network, then let it pop the switch dialog, then accept the switch, and which point the option to interact with all the existing switch options in queue start working again.

The key to reproducing this bug is to close the tab that asked you to switch instead of pressing the cancel button inside that tab.  Once that tab is closed, the switch queue becomes non interactable.

![image](https://user-images.githubusercontent.com/49324924/149640244-42ae5d91-4975-44a1-8faa-92c53ffeb9ff.png)

**Comment by ccajas** [comment]
Agreed with @daemianmack, it seems counterintuitive for Metamask to display the "switch the network" window for a tab that is _out of focus._ The user voluntary changed focus to a different tab with a website that uses a different network and is expected that they want that network.

Often I find myself switching back twice, because of the following:

Open tab with website using network A
MetaMask is on network A
Open tab with website using network B
Change to network B
Metamask pops up "switch the network" for A 
I click "Switch network" to A
I use the drop-down list to select and return to B

It may be appropriate to raise a separate issue for this behavior, as it is not expected of a user to update for an out-of-focus tab.
